### PR TITLE
proxy tls support one-way authentication

### PR DESCRIPTION
**Pull request by xiyichan**
#17073 
Signed-off-by: xiyichan <2863768433@qq.com>

### CODE REVIEW

File: docs/user_guides/tls_proxy.md [hunk 1]
```diff
@@ -429,6 +429,7 @@ They will creat 9 files about ca.key, ca.pem, ca.srl, server.key, server.pem, se
 ## Modify config
 Modify tlsEnabled to true and the file path in config/milvus.yaml.
```
Comment (jaime0815) [inline docs/user_guides/tls_proxy.md:430]: outdate description

File: docs/user_guides/tls_proxy.md [hunk 2]
```diff
@@ -429,6 +429,7 @@ They will creat 9 files about ca.key, ca.pem, ca.srl, server.key, server.pem, se
 ## Modify config
 Modify tlsEnabled to true and the file path in config/milvus.yaml.  
 Server need server.pem, server.key and ca.pem. Client need client.pem, client.key, ca.pem.
```
Comment (jaime0815) [inline docs/user_guides/tls_proxy.md:431]: We'd better add two parts to describe how to config:
- server-side
- client-side

File: internal/distributed/proxy/service_test.go [hunk 3]
```diff
@@ -792,114 +780,6 @@ func (m *MockProxy) ListCredUsers(ctx context.Context, req *milvuspb.ListCredUse
...
-var clientPemPath = "../../../configs/cert/client.pem"
-var clientKeyPath = "../../../configs/cert/client.key"
-
-// waitForServerReady wait for internal grpc service and external service to be ready, according to the params.
-func waitForServerReady() {
-	waitForGrpcReady(newWaitOption(waitDuration, Params.InternalPort, false, "", "", ""))
-	waitForGrpcReady(newWaitOption(waitDuration, Params.Port, Params.TLSEnabled, clientPemPath, clientKeyPath, Params.CaPemPath))
-}
-
-func runAndWaitForServerReady(server *Server) error {
```
Comment (czs007) [inline internal/distributed/proxy/service_test.go:894]: why  delete this ?

File: internal/distributed/proxy/service_test.go [hunk 4]
```diff
@@ -792,114 +780,6 @@ func (m *MockProxy) ListCredUsers(ctx context.Context, req *milvuspb.ListCredUse
...
-	ticker := time.NewTicker(opt.Duration)
-	ch := make(chan error, 1)
-
-	go func() {
-		// just used in UT to self-check service is available.
-		address := "localhost:" + strconv.Itoa(opt.Port)
-		if !opt.TLSEnabled {
-			_, err := grpc.Dial(address, grpc.WithBlock(), grpc.WithInsecure())
-			ch <- err
-		} else {
```
Comment (longjiquan) [inline internal/distributed/proxy/service_test.go:858]: Please also cover all cases that `tls_mode` in [0, 1, 2] to gracefully wait service ready.

**Pull request closed without merging**